instrumentation/jdbc/library/src/test/groovy/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java

```diff
@@ -0,0 +1,263 @@
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class OpenTelemetryDriverTest {
+
+  @DisplayName("verify driver auto registered")
+  @Test
+  void verifyOpenTelemetryDriverAutoRegisteration() throws ClassNotFoundException {
+
+    Enumeration<Driver> drivers = DriverManager.getDrivers();
+
+    // From JDBC 4.0 (Java SE 6), the driver can be auto-registered from the java.sql.Driver file
+    // contained in the META-INF.services folder
+    assertTrue(OpenTelemetryDriver.isRegistered());
+
+    assertThat(drivers.nextElement()).isEqualTo(OpenTelemetryDriver.INSTANCE);
+  }
+
+  @DisplayName("verify standard properties")
+  @Test
+  void verifyStandardProperties() {
+
+    assertThat(OpenTelemetryDriver.INSTANCE.jdbcCompliant()).isFalse();
+
+    String[] parts = Instrumenter.class.getPackage().getImplementationVersion().split("\\.");
+
+    assertThat(OpenTelemetryDriver.INSTANCE.getMajorVersion())
+        .isEqualTo(Integer.parseInt(parts[0]));
+    assertThat(OpenTelemetryDriver.INSTANCE.getMinorVersion())
+        .isEqualTo(Integer.parseInt(parts[1]));
+  }
+
+  @DisplayName("verify parent logger thrown an exception")
+  @Test
+  void verifyParentLoggerThrownAnException() {
+    assertThrows(
+        SQLFeatureNotSupportedException.class,
+        () -> OpenTelemetryDriver.INSTANCE.getParentLogger(),
+        "Feature not supported");
+  }
+
+  @DisplayName("verify accepted urls")
+  @ParameterizedTest
+  @MethodSource("provideInputOutputForAcceptUrl")
+  void verifyAcceptedUrls(String input, boolean expected) {
+    OpenTelemetryDriver driver = OpenTelemetryDriver.INSTANCE;
+    assertThat(driver.acceptsURL(input)).isEqualTo(expected);
+  }
+
+  private static Stream<Arguments> provideInputOutputForAcceptUrl() {
+    return Stream.of(
+        Arguments.of(null, false),
+        Arguments.of("", false),
+        Arguments.of("jdbc:", false),
+        Arguments.of("bogus:string", false),
+        Arguments.of("jdbc:postgresql://127.0.0.1:5432/dbname", false),
+        Arguments.of("jdbc:otel:postgresql://127.0.0.1:5432/dbname", true));
+  }
+
+  @DisplayName("verify connection with null url")
+  @Test
+  void verifyConnectionWithNullUrl() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> OpenTelemetryDriver.INSTANCE.connect(null, null),
+        "url is required");
+  }
+
+  @DisplayName("verify connection with empty url")
+  @Test
+  void verifyConnectionWithEmptyUrl() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> OpenTelemetryDriver.INSTANCE.connect(" ", null),
+        "url is required");
+  }
+
+  @DisplayName("verify connection with not accepted url")
+  @Test
+  void verifyConnectionWithNotAcceptedUrl() throws SQLException {
+    Connection connection = OpenTelemetryDriver.INSTANCE.connect("abc:xyz", null);
+    assertNull(connection);
+  }
+
+  @DisplayName("With other driver")
+  @Test
+  void verifyAddDriveCandidate() throws SQLException {
```

Comment:
What's the reason for merging several test cases into a very large one?

Reply:
can you see if `@Execution(SAME_THREAD)` on those tests solves the parallelization issue?

Reply:
You sure they ran in parallel? As far as I know we don't enable parallel test execution. Perhaps the problem is that theses tests run in a different order. Spock runs methods in the order they appear in the class but junit does not. See https://www.baeldung.com/junit-5-test-order for ideas how you could make tests run in predictable order.

Reply:
I have separated all the test cases, modified  and forced another one to execute at first.